targets/unity-v2/Testing/Tests/Entity/EntityApiTests.cs

```diff
@@ -141,6 +145,194 @@ private void GetObjectCallback2(GetObjectsResponse result)
+        {
+            var loginRequest = new ClientModels.LoginWithCustomIDRequest
+            {
+                CustomId = PlayFabSettings.BuildIdentifier,
+                CreateAccount = true,
+                LoginTitlePlayerAccountEntity = true,
+            };
+
+            PlayFabClientAPI.LoginWithCustomID(loginRequest, PlayFabUUnitUtils.ApiActionWrapper<ClientModels.LoginResult>(testContext, LoginCallbackPutTest), PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, SharedErrorCallback), testContext);
+        }
+        private void LoginCallbackPutTest(ClientModels.LoginResult result)
+        {
+            var testContext = (UUnitTestContext)result.CustomData;
+
+            if (result.EntityToken != null)
+            {
+                PlayFab.EntityModels.EntityTypes entityType = (PlayFab.EntityModels.EntityTypes)(int)result.EntityToken.Entity.Type; // possible loss of data 
+
+                _entityKey = new PlayFab.EntityModels.EntityKey
+                {
+                    Id = result.EntityToken.Entity.Id,
+                    Type = entityType,
+                    TypeString = result.EntityToken.Entity.TypeString
+                };
+
+                LoadFiles(testContext);
+            }
+            else
+            {
+                testContext.EndTest(UUnitFinishState.FAILED, "Entity Token is null!");
+            }
+        }
+        private void LoadFiles(UUnitTestContext testContext)
+        {
+            var request = new GetFilesRequest
+            {
+                Entity = _entityKey
+            };
+
+            PlayFabEntityAPI.GetFiles(request, PlayFabUUnitUtils.ApiActionWrapper<EntityModels.GetFilesResponse>(testContext, OnGetFilesInfo), PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, SharedErrorCallback), testContext);
+        }
+        void OnGetFilesInfo(GetFilesResponse result)
+        {
+            var testContext = (UUnitTestContext)result.CustomData;
+            bool testFileFound = false;
+            GetFileMetadata fileMetaData = new GetFileMetadata();
+
+            foreach (var eachFilePair in result.Metadata)
+            {
+                if (eachFilePair.Key.Equals(TEST_FILE_NAME))
+                {
+                    testFileFound = true;
+
+                    fileMetaData = eachFilePair.Value;
+                    break; // this test only support one file
+                }
+            }
+
+            if (!testFileFound)
+            {
+                UploadFile(TEST_FILE_NAME, testContext);
+            }
+            else
+            {
+                GetActualFile(fileMetaData, testContext);
+            }
+        }
+        void GetActualFile(GetFileMetadata fileData, UUnitTestContext testContext)
+        {
+            PlayFabHttp.SimpleGetCall(fileData.DownloadUrl,
+                result =>
+                {
+                    _testFileData = Encoding.UTF8.GetString(result);
```

Comment:
TestFileContent should just receive _testFileData as a parameter, and it should not be saved to a static variable like this. Use a local var, and pass it into TestFileContent.
This is a less severe kind of potential test-leakage that I was talking about with _testContext.

Reply:
Modified